andela/wg-rogue-one
# feat(upgrade-nutrition-plan) Upgrade nutrition plans

### What does this PR do?
This PR adds a feature that allows a user to add a meal that he/she has
already eaten in addition to adding the meals he/she plans to eat.

Add meal_category options to MealItem model in `wger/nutrition/models.py`
Add migrations file `0005_mealitem_meal_category.py`
Add meal_category select element to `wger/nutrition/templates/meal_item/edit.html`
Update `test_meal_item` and `test_meal`
Add meal time validation to `wager-core.js`

#### Description of Task to be completed?
Add support for dropsets((https://en.wikipedia.org/wiki/Drop_set)

#### How should this be manually tested?
1: Clone the application and install the requirements in your virtual environment.
2: Update the database settings in the `settings.py` file with your postgres user, password and database name.
3: Run the application with `python manage.py runserver`.
4: Register and login.
5: Go to Nutrition -> Nutrition plans.
6: Click `Add a meal`

7: You should get a modal with the option to add a meal as a new meal or previous meal. Below is a sample image
<img width="1144" alt="screenshot 2018-06-21 at 22 54 36" src="https://user-images.githubusercontent.com/25340897/41742354-62edbb98-75a6-11e8-8acf-ceedfc63e658.png">

8: Save. Below is a sample image of the view of the meal plans
<img width="1030" alt="screenshot 2018-06-21 at 22 54 14" src="https://user-images.githubusercontent.com/25340897/41742413-8910962e-75a6-11e8-9560-9fa1ed112bda.png">

#### Any background context you want to provide?
This feature was not available.

#### What are the relevant pivotal tracker stories?
#157731619
